packages/hash/frontend/src/blocks/page/createEditorView.ts

```diff
@@ -21,6 +31,85 @@ import { createTextEditorView } from "./createTextEditorView";
 
 export type BlocksMap = Record<string, HashBlock>;
 
+const createSavePlugin = (
+  ownedById: string,
+  pageEntityId: string,
+  client: ApolloClient<unknown>,
+) => {
+  let saveQueue = Promise.resolve<unknown>(null);
+
+  const triggerSave = (view: EditorView<Schema>) => {
+    saveQueue = saveQueue.catch().then(async () => {
+      const [newContents, newDraftToEntityId] = await save(
+        client,
+        ownedById,
+        pageEntityId,
+        view.state.doc,
+        entityStorePluginState(view.state).store,
+      );
+
+      const { tr } = view.state;
+      addEntityStoreAction(view.state, tr, {
+        type: "mergeNewPageContents",
+        payload: {
+          blocks: newContents,
+          presetDraftIds: newDraftToEntityId,
+        },
+      });
+
+      view.updateState(view.state.apply(tr));
+    });
+  };
+
+  const minWaitTime = 400;
+  const maxWaitTime = 1200;
+
+  // Saving happens through a debounced write operation
+  const writeDebounce = debounce(
+    (view: EditorView<Schema>) => {
+      triggerSave(view);
+    },
+    minWaitTime,
+    { maxWait: maxWaitTime },
+  );
+
+  return new Plugin<unknown, Schema>({
+    view: (_viewOnCreation: EditorView<Schema>) => {
+      return {
+        update: (view, prevState) => {
+          if (view.state.doc !== prevState.doc) {
+            // If the document changes between updates, we issue a debounced update.
+            writeDebounce(view);
+          }
```

Comment:
What about if the entity store changes?

Reply:
Debounce + interval saving in b6eec1eabd5440c1aaccf4f979b7e1c98daf2b87 (seems to be a little more stable with this approach)

Reply:
You can detect entity store changes – plugin state is stored in EditorState.

Reply:
I see, thank you. If this results in problems down the line, we can address it in a follow up